File: src/Solutions/Euler291.hs
module Solutions.Euler291 where
import Math.NumberTheory.Primes.Testing(isPrime)

-- https://oeis.org/A027862
euler291 :: IO String
euler291 = return $ show $ length $ filter isPrime $ takeWhile (<5 * (10^15)) $ map (\n -> n^2 + (n + 1)^2) [1..]

File: rust/src/solutions/euler291.rs
use primal::is_prime;

pub fn euler291() -> String {
    (1u64..)
        .map(|n| n * n + (n + 1) * (n + 1))
        .take_while(|&n| n < 5 * 10u64.pow(15))
        .filter(|&n| is_prime(n))
        .count()
        .to_string()
}
